### ISSUE TITLE
Fail to create an ACI context in Azure China/Azure Germany account

### ISSUE THREAD

**Issue by v-ruizh** [microsoft/vscode-docker#2775]
**OS:** Win 10
**Build Version:** 20210312.1
**Docker Desktop:** 3.2.1 (61626)
![image](https://user-images.githubusercontent.com/45614674/111123495-5dd35e80-85aa-11eb-8824-3f63a5071697.png)

**Repro Steps:**
1. Sign in with Azure China account.
2. Try to create an ACI context.

**Expect:**
Succeed to create an ACI context.

**Actual:**
Fail to create an ACI context.
```
Creating ACI context "serere-ch1"...
Error: Process 'docker context create aci serere-ch1 --subscriptio...' exited with code 1
Error: could not find the requested subscription from your Azure login. You might need to specify a tenant ID with docker login azure --tenant-id xxx 
```
**More Info:**
This issue also reproduces in Azure Germany account.

**Comment by bwateratmsft** [comment]
Looks like ACR isn't supported in Germany: https://azure.microsoft.com/en-us/pricing/details/container-registry/

![image](https://user-images.githubusercontent.com/36966225/111173403-0074e880-857d-11eb-856d-59eff8cf6415.png)

**Comment by bwateratmsft** [comment]
Closed the wrong item.

**Comment by bwateratmsft** [comment]
Looks like ACI _is_ supported in both Germany and China. @karolz-ms do you know what we might need to change to get this working there? Looks like we may at least need to specify a tenant ID?

**Comment by karolz-ms** [comment]
Ah, yes, I should have opened an issue about this. When working with clouds other than Azure public cloud, we will need to specify `--cloud-name` when doing `docker login azure`. In case of Azure China it is "AzureChinaCloud".

```shell
C:\code>docker login azure --help
Log in to azure

Usage:
  docker login azure [flags]

Flags:
      --client-id string       Client ID for Service principal login
      --client-secret string   Client secret for Service principal login
      --cloud-name string      Name of a registered Azure cloud
  -h, --help                   Help for azure
      --tenant-id string       Specify tenant ID to use
```

The list of well-known clouds is defined here https://github.com/docker/compose-cli/blob/main/aci/login/cloud_environment.go#L166 but there might be other clouds than the well-known ones, so we should allow the user to type a name too, in addition to giving them the well-known choices.

**Comment by bwateratmsft** [comment]
This fix is now released with Docker extension version 1.12.0.
